backend/routing/QuizRoutes.go

```diff
@@ -3,22 +3,35 @@ package routing
 import (
 	"encoding/json"
 	"net/http"
-
-	"github.com/fourtf/studyhub/utils"
 )
 
 type apiResponse struct {
-	Error string      `json:"error"`
-	Data  interface{} `json:"data"`
+	Data interface{} `json:"data"`
 }
 
-type quiz struct {
+const (
+	multipleCoiceKey = "multiple-coice"
+	freeTextKey      = "free-text"
+)
+
+type question struct {
 	ID string `json:"id"`
-	// single | multi | freetext
-	Type     string   `json:"type"`
-	Question string   `json:"question"`
-	Answer   *string  `json:"answer,omitifempty"`
-	Answers  []answer `json:"answers,omitifempty"`
+	// multiple-choice | freetext
+	Type string      `json:"type"`
+	Data interface{} `json:"data"`
+}
+
+// single answer -> radio button
+type multipleCoiceData struct {
+	AllowMultiple bool     `json:"allowMultiple"`
+	Question      string   `json:"question"`
+	Answers       []answer `json:"answers"`
+}
+
+// full test answers -> text input
+type freeTextData struct {
+	Question string `json:"question"`
+	Answer   string `json:"answer"`
 }
```

Comment:
With this solution we are still able to set the QuestionType to freetext but assign a multiple choice question. I thought of something like this:
```suggestion
type question struct {
  ID string `json: "id"`
  Question string `json: "question"`
}

type multipleChoiceQuestion struct {
 question //inherits fields ID and Question
 HasMultipleCorrectAnswers bool `json: "hasMultipleCorrectAnswers"`
 Answers []answer `json: "answers"`
}

type freeTextQuestion struct {
 question //inherits fields ID and Question
 Answer string `json: "answer"`
}
```
->Now we're not able to accidentally create invalid questions, but we're still able to tell which type of question we have by using an instanceof-check in the frontend

Reply:
I checked if my recommended solution works. It is not possible to instanceof-check, but we can check if a property is in our object do decide which cast we have to perform. For example we could check if the property hasMultipleCorrectAnswers is in our question object and then perform the cast. It is not necessary to send the type information in this case which allows a more simple structure of our objects.